### PR TITLE
week-4-assessment-leo

### PR DESCRIPTION
**Pull request by LeoContreras93**
# Submitting Week 4 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x] JavaScript Coding Challenges
- [x] Ruby Coding Challenges
- [x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,6 +17,25 @@
...
 
+// describe('removeColor', () => {
+//     const expected = (["purple", "blue", "green", "yellow", "pink"], ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"] );
+//     it('takes in an array, removes the first item from the array and shuffles the remaining content.', () => {
+//       expect(colors1).toEqual(expect.arrayContaining(expected));
+//     });
+//     it('takes in an array, removes the first item from the array and shuffles the remaining content.', () => {
+//       expect(colors2).not.toEqual(expect.arrayContaining(expected));
+//     });
+//   });
```
Comment (elysemontano) [inline code-challenges.test.js:28]: Please make sure to leave your test uncommented out.  It looks like you are really close on this test, just keep in mind that each test variable will have a separate expected outcome.

File: code-challenges.test.js [hunk 2]
```diff
@@ -23,27 +44,117 @@ const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "s
...
+//     let firstElement = array.slice(1)
+//     return array
+// }
+
+const removeColor = (array) => {
+    let newArray = array.slice(1)
+        return newArray.sort()
+    
+// console.log(removeColor(colors1))
+//unable to figure this one out.
```
Comment (elysemontano) [inline code-challenges.test.js:69]: You are really close on this one!  You are using. the right built-in methods to get this working, the sort method needs just a little more information though.  Highly recommend trying to google sort randomly in Javascript.  This [article](https://dev.to/codebubb/how-to-shuffle-an-array-in-javascript-2ikj) may be helpful.

File: code-challenges.test.js [hunk 3]
```diff
@@ -23,27 +44,117 @@ const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "s
...
+//name function numbersArray
+//iterate over the array using map
+//use math.max and math.min to return largest and lowest values
+
+// const numbersArray = (array) => {
+//     let minMath = [Math.min(...array)]
+//     let maxMath = [Math.max(...array)]
+//     return minMath.concat(maxMath)
+    
+// }
```
Comment (elysemontano) [inline code-challenges.test.js:103]: This looks great!  Please provide pseudocode for each variation you try, and leave the one that is closest to where you want to be uncommented out.

File: code-challenges.test.js [hunk 4]
```diff
@@ -23,27 +44,117 @@ const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "s
...
+//     array.map
+//     return (Math.max(Math.min(array)))
+ 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
+// describe("randomArray", () => {
+//     it("takes in two arrays as arguments and returns one array with no duplicate values.", () => {
+//         expect(randomArray(testArray1)).toEqual([3, 7, 10, 5, 4, 8, 2, 1])
```
Comment (elysemontano) [inline code-challenges.test.js:128]: You will likely need to pass both test variables as an argument here.

File: code-challenges.test.js [hunk 5]
```diff
@@ -23,27 +44,117 @@ const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "s
...
+
 const testArray1 = [3, 7, 10, 5, 4, 3, 3]
 const testArray2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
 
 
 // b) Create the function that makes the test pass.
+//name function randomArray
+//takes in two arguments 
+//returns 1 array with no duplicate numbers
```
Comment (elysemontano) [inline code-challenges.test.js:141]: I really need to see more pseudocode in the future that includes the methods you are considering implementing in your code.

File: interview-questions.md [hunk 6]
```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: props is information that is only passed in one direction from parent to child component. They are snapshot of state.
 
-  Researched answer:
+  Researched answer: props gives us the ability to communicate by passing data & methods between  components or to pass data down to nested components. As I mention before they are a snapshot of state that are passed on to components tasked with displaying & or letting a user interact with that information. the information is shared through the component call.
```
Comment (elysemontano) [inline interview-questions.md:12]: I like the wording here that props are a snapshot of state.

File: interview-questions.md [hunk 7]
```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: Document Object Model are user interactions by the user on a web application, whether its clicking with a mouse or using your keyboard to type.
 
-  Researched answer:
+  Researched answer: interactions like mouse clicking, hovering or typing with a keyboard on your web application are all user interactions in where the the DOM is listening for these changes. As you interact with your web page for example using a mouse the DOM will react according to the position of your cursor and respond accordingly.
```
Comment (elysemontano) [inline interview-questions.md:20]: Careful to not confuse the DOM with the Virtual DOM.  The DOM is a visual representation of the markup and code logic that makes up a website.

File: interview-questions.md [hunk 8]
```diff
@@ -7,52 +7,52 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: integers can be negative or positive numbers. Integers can be stored in variables and print the value by just referencing the variable name. Floats you can add integers and floats together and return a float.
 
 
 
 5. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: returns the last line of code in a block.
```
Comment (elysemontano) [inline interview-questions.md:42]: Make sure that you are answering it exactly how you would in an interview.  That means, complete sentence and expanding on the concept as thoroughly as possible.

File: ruby-challenges.rb [hunk 9]
```diff
@@ -10,6 +10,17 @@
 num3 = 221
 # Expected output: '221 is odd'
 
+def even_odd (number)
+    if number % 2 == 0
+        "number is even"
+    else
+        "number is odd"
+    end
+end
```
Comment (elysemontano) [inline ruby-challenges.rb:19]: Nice!  Take a look at the Ruby methods .even? and .odd? when you get the chance.

**Review by elysemontano — APPROVED** [review]
Awesome work on the Ruby section Leo!  Your logic on the code section is heading in the right direction, and tests look good, just make sure to leave it uncommented out.  On the next few assessments, I do need to see more pseudocode breaking down the code that you have.  Methods, iterations, higher-order functions, and anything that has logic that you are considering in your code should be included in your pseudocode.

File: code-challenges.test.js (hunk 2 above)
Comment (LeoContreras93) [reply code-challenges.test.js:69]: thank you I will review the article.

**Comment by LeoContreras93** [comment]
Thank you for the overall feedback. I will work on my pseudocode.

**Pull request merged**